### PR TITLE
pydantic model for recommendations

### PR DESCRIPTION
**Pull request by vansika**
One of the users got API Internal Server Error while trying to get the recommended tracks on listenbrainz.org. I checked the API endpoint and realised that recommendations for that user weren't inserted into the DB in the right format. Defining a pydantic model for recommendations will make sure that data received from spark which isn't in the format as expected is not be inserted into the DB.

### CODE REVIEW

File: listenbrainz_spark/recommendations/recommend.py [hunk 1]
```diff
@@ -277,31 +281,78 @@ def create_messages(top_artist_rec_mbid_df, similar_artist_rec_mbid_df, active_u
...
+                current_app.logger.warning("""Invalid entry present in top artist recommendations for user: {user_name},
+                                           skipping""".format(user_name=row.user_name), exc_info=True)
+
             user_rec[row.user_name]['similar_artist'] = []
 
         else:
-            user_rec[row.user_name]['top_artist'].append([row.mb_recording_mbid, row.rating])
+            try:
+                user_rec[row.user_name]['top_artist'].append(
+                    UserRecommendationsRecord(**
```
Comment (paramsingh) [inline listenbrainz_spark/recommendations/recommend.py:300]: why are we expecting an error here? going from dict -> pydantic object -> dict seems redundant. maybe it makes sense to just keep a dict here

File: listenbrainz/spark/handlers.py [hunk 2]
```diff
@@ -304,11 +306,20 @@ def handle_recommendations(data):
...
-    top_artist_recording_mbids = data['top_artist']
-    similar_artist_recording_mbids = data['similar_artist']
+    recommendations = data['recommendations']
 
-    db_recommendations_cf_recording.insert_user_recommendation(user['id'], top_artist_recording_mbids,
-                                                               similar_artist_recording_mbids)
+    try:
+        db_recommendations_cf_recording.insert_user_recommendation(
+            user['id'],
+            UserRecommendationsJson(**{
```
Comment (paramsingh) [inline listenbrainz/spark/handlers.py:314]: :tada:

**Review by paramsingh — APPROVED** [review]
Changes make sense to me, I'm wondering if we really need the conversion on the spark side before sending the message. it seems a bit redundant, wdyt?

File: listenbrainz_spark/recommendations/recommend.py (hunk 1 above)
Comment (vansika) [reply listenbrainz_spark/recommendations/recommend.py:300]: Hmm. Can you merge and restart request consumer for now? I will update this later.  I guess you won't be available in the morning to restart request consumer, and I want to test stuff for a user whose recommend tracks page raise ISE.

Comment (paramsingh) [reply listenbrainz_spark/recommendations/recommend.py:300]: Done! Can you open a ticket (or even better, a PR) addressing the comment?

**Pull request merged**